src/natcap/invest/urban_nature_access.py

```diff
@@ -254,103 +282,133 @@ def execute(args):
-        dependent_task_list=[lulc_alignment_task]
-    )
-
-    search_radius_in_pixels = abs(
-        float(args['search_radius']) / squared_lulc_pixel_size[0])
-    kernel_creation_task = graph.add_task(
-        # All kernel creation types have the same function signature
-        kernel_types[kernel_type],
-        args=(search_radius_in_pixels, file_registry['kernel']),
-        kwargs={'normalize': False},  # Model math calls for un-normalized
-        task_name=f'2SFCA - Create {kernel_type} kernel',
-        target_path_list=[file_registry['kernel']]
-    )
-
-    convolved_population_task = graph.add_task(
-        _convolve_and_set_lower_bounds_for_population,
+    preprocess_attr_table_task = graph.add_task(
+        _preprocess_lulc_attribute_table,
         kwargs={
-            'signal_path_band': (file_registry['aligned_population'], 1),
-            'kernel_path_band': (file_registry['kernel'], 1),
-            'target_path': file_registry['convolved_population'],
-            'working_dir': intermediate_dir,
+            'source_attr_table_path': args['lulc_attribute_table'],
+            'default_radius': float(args['search_radius']),
+            'target_attr_table_path': file_registry['attribute_table'],
         },
-        task_name='2SFCA - Convolve population',
-        target_path_list=[file_registry['convolved_population']],
-        dependent_task_list=[
-            kernel_creation_task,
-            population_alignment_task,
-        ])
-
-    greenspace_nodata = pygeoprocessing.get_raster_info(
-        file_registry['greenspace_area'])['nodata'][0]
-    population_nodata = pygeoprocessing.get_raster_info(
-        file_registry['convolved_population'])['nodata'][0]
-    greenspace_population_ratio_task = graph.add_task(
-        pygeoprocessing.raster_calculator,
-        kwargs={
-            'base_raster_path_band_const_list': [
-                (file_registry['greenspace_area'], 1),
-                (file_registry['convolved_population'], 1),
-                (greenspace_nodata, 'raw'),
-                (population_nodata, 'raw')],
-            'local_op': _greenspace_population_ratio,
-            'target_raster_path': file_registry['greenspace_population_ratio'],
-            'datatype_target': gdal.GDT_Float32,
-            'nodata_target': FLOAT32_NODATA
-        },
-        task_name='2SFCA: Calculate R_j greenspace/population ratio',
-        target_path_list=[file_registry['greenspace_population_ratio']],
-        dependent_task_list=[
-            convolved_population_task,
-            greenspace_reclassification_task,
-        ])
-
-    convolved_greenspace_population_ratio_task = graph.add_task(
-        pygeoprocessing.convolve_2d,
+        target_path_list=[file_registry['attribute_table']],
+        priority=10,  # higher-than-normal priority to address sooner.
+        task_name='Preprocess the attribute table')
+
+    preprocess_attr_table_task.join()
```

Comment:
Does setting the `priority=10` still have any effect if we `join` the task first?

Reply:
In this case, the issue is addressed by no longer needing to preprocess the table since the `search_radius_m` column is now required.

In the general case, though, yes, setting `priority=10` could still have an effect when the task it `join`ed.  Imagine a case where `n_workers > 0`, which means that the order of task execution is not guaranteed.  It's possible for us to get into a case where all of the other tasks that were added other than this one task to execute before `preprocess_attr_table_task`.  Increasing the priority means that this task will take priority if another lower-priority task also has satisfied its dependencies.